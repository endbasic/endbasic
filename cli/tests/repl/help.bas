' EndBASIC
' Copyright 2022 Julio Merino
'
' Licensed under the Apache License, Version 2.0 (the "License"); you may not
' use this file except in compliance with the License.  You may obtain a copy
' of the License at:
'
'     http://www.apache.org/licenses/LICENSE-2.0
'
' Unless required by applicable law or agreed to in writing, software
' distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
' WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
' License for the specific language governing permissions and limitations
' under the License.

' Dumps all help messages for validation purposes.
'
' The interactive help messages are built into the interpreter as multiline
' raw strings and it is too easy to make mistakes when typing them (especially
' around line boundaries).  This dump serves to sanity-check that formatting
' works as intended.

PRINT "Output from HELP:"
HELP

DO
    READ topic$
    IF topic$ = "__DONE__" THEN EXIT DO
    PRINT "Output from HELP \""; topic$; "\":"
    HELP topic$
LOOP

' Help topics.
DATA "ARRAY"
DATA "CLOUD"
DATA "CONSOLE"
DATA "DATA"
DATA "FILE SYSTEM"
DATA "GRAPHICS"
DATA "HARDWARE"
DATA "INTERPRETER"
DATA "LANG"
DATA "NUMERICAL"
DATA "STORED"
DATA "STRING"

' Language reference.
DATA "ASSIGNMENTS"
DATA "ERROR HANDLING"
DATA "EXPRESSIONS"
DATA "FLOW CONTROL"
DATA "MISCELLANEOUS"
DATA "SYMBOLS"

' Commands.
DATA "CD"
DATA "CLEAR"
DATA "CLS"
DATA "COLOR"
DATA "DEG"
DATA "DIR"
DATA "EDIT"
DATA "GFX_CIRCLE"
DATA "GFX_CIRCLEF"
DATA "GFX_LINE"
DATA "GFX_PIXEL"
DATA "GFX_RECT"
DATA "GFX_RECTF"
DATA "GFX_SYNC"
DATA "GPIO_CLEAR"
DATA "GPIO_SETUP"
DATA "GPIO_WRITE"
DATA "HELP"
DATA "INPUT"
DATA "KILL"
DATA "LIST"
DATA "LOAD"
DATA "LOCATE"
DATA "LOGIN"
DATA "LOGOUT"
DATA "MOUNT"
DATA "NEW"
DATA "PRINT"
DATA "PWD"
DATA "RAD"
DATA "RANDOMIZE"
DATA "READ"
DATA "RESTORE"
DATA "RUN"
DATA "SAVE"
DATA "SHARE"
DATA "SIGNUP"
DATA "SLEEP"
DATA "UNMOUNT"

' Functions.
DATA "ASC"
DATA "ATN"
DATA "CHR"
DATA "CINT"
DATA "COS"
DATA "ERRMSG"
DATA "GFX_HEIGHT"
DATA "GFX_WIDTH"
DATA "GPIO_READ"
DATA "INKEY"
DATA "INT%"
DATA "LBOUND"
DATA "LEFT"
DATA "LEN"
DATA "LTRIM"
DATA "MAX"
DATA "MID"
DATA "MIN"
DATA "PI"
DATA "RIGHT"
DATA "RND"
DATA "RTRIM"
DATA "SCRCOLS"
DATA "SCRROWS"
DATA "SIN"
DATA "SQR"
DATA "STR$"
DATA "TAN"
DATA "UBOUND"

' End of data marker.
DATA "__DONE__"
