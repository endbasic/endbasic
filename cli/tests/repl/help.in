' EndBASIC
' Copyright 2020 Julio Merino
'
' Licensed under the Apache License, Version 2.0 (the "License"); you may not
' use this file except in compliance with the License.  You may obtain a copy
' of the License at:
'
'     http://www.apache.org/licenses/LICENSE-2.0
'
' Unless required by applicable law or agreed to in writing, software
' distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
' WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
' License for the specific language governing permissions and limitations
' under the License.

' Dumps all help messages for validation purposes.
'
' The interactive help messages are built into the interpreter as multiline
' raw strings and it is too easy to make mistakes when typing them (especially
' around line boundaries).  This dump serves to sanity-check that formatting
' works as intended.

'
' General help topics.
'

PRINT "Output from HELP:"
HELP

PRINT "Output from HELP ARRAY:"
HELP ARRAY

PRINT "Output from HELP CLOUD:"
HELP CLOUD

PRINT "Output from HELP CONSOLE:"
HELP CONSOLE

PRINT "Output from HELP \"FILE SYSTEM\":"
HELP "FILE SYSTEM"

PRINT "Output from HELP HARDWARE:"
HELP HARDWARE

PRINT "Output from HELP INTERPRETER:"
HELP INTERPRETER

PRINT "Output from HELP LANG:"
HELP LANG

PRINT "Output from HELP NUMERICAL:"
HELP NUMERICAL

PRINT "Output from HELP STORED:"
HELP STORED

PRINT "Output from HELP STRING:"
HELP "STRING"

'
' Help on commands.
'

PRINT "Output from HELP CD:"
HELP CD

PRINT "Output from HELP CLEAR:"
HELP CLEAR

PRINT "Output from HELP CLS:"
HELP CLS

PRINT "Output from HELP COLOR:"
HELP COLOR

PRINT "Output from HELP DEL:"
HELP DEL

PRINT "Output from HELP DIR:"
HELP DIR

PRINT "Output from HELP EDIT:"
HELP EDIT

PRINT "Output from HELP EXIT:"
HELP EXIT

PRINT "Output from HELP GPIO_CLEAR:"
HELP GPIO_CLEAR

PRINT "Output from HELP GPIO_SETUP:"
HELP GPIO_SETUP

PRINT "Output from HELP GPIO_WRITE:"
HELP GPIO_WRITE

PRINT "Output from HELP HELP:"
HELP HELP

PRINT "Output from HELP INPUT:"
HELP INPUT

PRINT "Output from HELP LIST:"
HELP LIST

PRINT "Output from HELP LOAD:"
HELP LOAD

PRINT "Output from HELP LOCATE:"
HELP LOCATE

PRINT "Output from HELP LOGIN:"
HELP LOGIN

PRINT "Output from HELP MOUNT:"
HELP MOUNT

PRINT "Output from HELP NEW:"
HELP NEW

PRINT "Output from HELP PWD:"
HELP PWD

PRINT "Output from HELP PRINT:"
HELP PRINT

PRINT "Output from HELP RANDOMIZE:"
HELP RANDOMIZE

PRINT "Output from HELP RUN:"
HELP RUN

PRINT "Output from HELP SAVE:"
HELP SAVE

PRINT "Output from HELP SHARE:"
HELP SHARE

PRINT "Output from HELP SLEEP:"
HELP SLEEP

PRINT "Output from HELP UNMOUNT:"
HELP UNMOUNT

'
' Help on functions.
'

PRINT "Output from HELP DTOI:"
HELP DTOI

PRINT "Output from HELP GPIO_READ:"
HELP GPIO_READ

PRINT "Output from HELP ITOD:"
HELP ITOD

PRINT "Output from HELP LEFT:"
HELP LEFT

PRINT "Output from HELP LEN:"
HELP LEN

PRINT "Output from HELP LBOUND:"
HELP LBOUND

PRINT "Output from HELP LTRIM:"
HELP LTRIM

PRINT "Output from HELP MID:"
HELP MID

PRINT "Output from HELP RIGHT:"
HELP RIGHT

PRINT "Output from HELP RND:"
HELP RND

PRINT "Output from HELP RTRIM:"
HELP RTRIM

PRINT "Output from HELP UBOUND:"
HELP UBOUND
