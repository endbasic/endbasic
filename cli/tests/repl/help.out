
    Welcome to EndBASIC X.Y.Z.
    Type HELP for interactive usage information.

Output from HELP:

    This is EndBASIC X.Y.Z.

    CLEAR     Clears all variables to restore initial state.
    CLS       Clears the screen.
    COLOR     Sets the foreground, the background, or both colors.
    DIR       Displays the list of files on disk.
    EDIT      Edits the stored program.
    EXIT      Exits the interpreter.
    HELP      Prints interactive help.
    INPUT     Obtains user input from the console.
    LIST      Lists the contents of the stored program.
    LOAD      Loads the given program.
    LOCATE    Moves the cursor to the given position.
    NEW       Clears the stored program from memory.
    PRINT     Prints a message to the console.
    RENUM     Reassigns line numbers to make them all multiples of ten.
    RUN       Runs the stored program.
    SAVE      Saves the current program in memory to the given filename.

    Type HELP followed by a command name for details on that command.

Output from HELP CLEAR:

    CLEAR

    Clears all variables to restore initial state.

Output from HELP CLS:

    CLS

    Clears the screen.

Output from HELP COLOR:

    COLOR [fg%][, bg%]

    Sets the foreground, the background, or both colors.

    Color numbers are given as ANSI numbers and can be between 0 and 255.  If a color number is not specified, then the color is left unchanged.

Output from HELP DIR:

    DIR

    Displays the list of files on disk.

Output from HELP EDIT:

    EDIT [lineno%]

    Edits the stored program.

    Without a line number, starts interactive editing at the last line of the stored program (or the first line if there is no program yet).  Editing ends on the first empty line.

    With a line number, and if the line does not yet exist, enters interactive editing for that line only.  Otherwise, if the line already exists, presents the contents of that line for interactive editing.  Editing terminates upon an ENTER key press.

Output from HELP EXIT:

    EXIT [code%]

    Exits the interpreter.

    The optional code indicates the return value to return to the system.

Output from HELP HELP:

    HELP [commandname]

    Prints interactive help.

    Without arguments, shows a summary of all available commands.

    With a single argument, shows detailed information about the given command.

Output from HELP INPUT:

    INPUT ["prompt"] <;|,> variableref

    Obtains user input from the console.

    The first expression to this function must be empty or evaluate to a string, and specifies the prompt to print.  If this first argument is followed by the short `;` separator, the prompt is extended with a question mark.

    The second expression to this function must be a bare variable reference and indicates the variable to update with the obtained input.

Output from HELP LIST:

    LIST

    Lists the contents of the stored program.

Output from HELP LOAD:

    LOAD filename

    Loads the given program.

    The filename must be a string and must be a basename (no directory components).  The .BAS extension is optional, but if present, it must be .BAS.

Output from HELP LOCATE:

    LOCATE row%, column%

    Moves the cursor to the given position.

Output from HELP NEW:

    NEW

    Clears the stored program from memory.

Output from HELP PRINT:

    PRINT [expr1 [<;|,> .. exprN]]

    Prints a message to the console.

    The expressions given as arguments are all evaluated and converted to strings.  Arguments separated by the short `;` separator are concatenated with a single space, while arguments separated by the long `,` separator are concatenated with a tab character.

Output from HELP RENUM:

    RENUM

    Reassigns line numbers to make them all multiples of ten.

Output from HELP RUN:

    RUN

    Runs the stored program.

    Note that the program runs in the context of the interpreter so it will pick up any variables and other state that may already be set.

Output from HELP SAVE:

    SAVE filename

    Saves the current program in memory to the given filename.

    The filename must be a string and must be a basename (no directory components).  The .BAS extension is optional, but if present, it must be .BAS.

End of input by CTRL-D
